sycl/include/CL/sycl/multi_ptr.hpp

```diff
@@ -516,12 +516,6 @@ class multi_ptr<const void, Space> {
         static_cast<elem_pointer_t>(m_Pointer));
   }
 
-  // Implicit conversion to multi_ptr<const void, Space>
-  operator multi_ptr<const void, Space>() const {
-    using ptr_t = typename detail::PtrValueType<const void, Space>::type *;
-    return multi_ptr<const void, Space>(reinterpret_cast<ptr_t>(m_Pointer));
-  }
-
```

Comment:
I reckon we should have a test checking conversion to const-void-multi_ptr.

Reply:
Agree, but how is this related to this patch?

Reply:
Thing is, this exact signature is in SYCL specification 1.2.1.

Reply:
It's a bug in the spec.

```
// Specialization of multi_ptr for void and const void
// VoidType can be either void or const void
template <access::address_space Space>
class multi_ptr<VoidType, Space> {
//...
// Implicit conversion to multi_ptr<const void, Space>
operator multi_ptr<const void, Space>() const;
//...
};
```

The operator should exist when `VoidType` is `void`, but not exist when `VoidType` is `const void`.  I'll report it.

Reply:
Thanks!